smach_viewer/scripts/smach_viewer.py

```diff
@@ -645,7 +645,9 @@ def __init__(self):
         # Wait till we have received the first structure message with a top container
         with self._update_cond:
             while not self._top_containers:
-                self._update_cond.wait()
+                self._update_cond.wait(5.0)
```

Comment:
```suggestion
                if not self._update_cond.wait(5.0):
```
You should only print the message if the [timeout occured](https://docs.python.org/3/library/threading.html#threading.Condition.wait).